src/libraries/System.Net.Http/src/System/Net/Http/Headers/NameValueHeaderValue.cs

```diff
@@ -358,7 +358,8 @@ private static void CheckValueFormat(string? value)
             }
 
             // Trailing/leading space are not allowed
-            if (value[0] == ' ' || value[0] == '\t' || value[^1] == ' ' || value[^1] == '\t')
+            if (value.StartsWith(' ') || value.StartsWith('\t')
+                || value.EndsWith(' ') || value.EndsWith('\t'))
```

Comment:
Formatting seems a bit strange.
```suggestion
            if (value.StartsWith(' ') || value.StartsWith('\t') || value.EndsWith(' ') || value.EndsWith('\t'))
```

Is not longer than the comment at the start of the method.